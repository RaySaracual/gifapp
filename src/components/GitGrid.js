import React from 'react'
import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GitGrid = ({ category }) => {

    const  { data:images, loading } = useFetchGifs( category );

    return (
        <>
        <h3> {category} </h3>
        {loading &&  <span className="animate__animated animate__flash"> Cargando... </span>}
        <div className="card-grid">
            {
                images.map(img => (
                    <GifGridItem 
                    key = {img.id}
                    {...img} />
                ))
            }

        </div>
        </>
    )
}
