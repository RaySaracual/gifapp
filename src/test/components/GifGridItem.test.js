import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Componente gitGridItem', () => {

    const title ='Hola';
    const url = 'http://localhost/hola.img';
    const wrapper  = shallow ( <GifGridItem title={title} url={url} /> );

    test('Debe mostrar <GifGridItem /> correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim()).toBe( title );
    })

    test('Debe tener la imagen igual al url  y alt de los props', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe (url);
        expect(img.prop('alt')).toBe (title);
        
    })
    

test('debe de tener animate__fadeIn    ', () => {

    const div = wrapper.find('div');
    const data = div.props().className;
  
    expect(data.includes('animate__fadeIn')).toBe(true);
    
})


    
})
