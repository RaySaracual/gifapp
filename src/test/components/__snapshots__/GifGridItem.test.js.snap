// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente gitGridItem Debe mostrar <GifGridItem /> correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hola"
    src="http://localhost/hola.img"
  />
  <p>
    Hola
  </p>
</div>
`;
